handlers/rsa.go

```diff
@@ -0,0 +1,222 @@
+		log.Warnf("invalid signature: %v", err)
+		return false, "invalid signature"
+	}
+
+	if st == csp.SignatureTypeSharedKey {
+		return true, "please, do not share the key"
+	}
+	if rh.exist(authData.VoterId) {
+		log.Warnf("%s already registered", authData.VoterId)
+		return false, "already registered"
+	}
+
+	err = rh.addKey(authData.VoterId, nil)
+	if err != nil {
+		log.Warnf("could not add key %x", authData.VoterId)
+		return false, "could not add key"
+	}
+	log.Infof("new user registered with id %s", authData.VoterId)
+
+	return true, ""
+}
+
+// RequireCertificate must return true if the auth handler requires some kind of client
+// TLS certificate. If true then CertificateCheck() and HardcodedCertificate() methods
+// must be correctly implemented. Else both function can just return true and nil.
+func (rh *RsaHandler) RequireCertificate() bool {
+	return false
+}
+
+// CertificateCheck is used by the Auth handler to ensure a specific certificate is
+// added to the CA cert pool on the HTTP/TLS layer (optional).
+func (rh *RsaHandler) CertificateCheck(subject []byte) bool {
+	return true
+}
+
+// Certificates returns a hardcoded CA certificated that will be added to the
+// CA cert pool by the handler (optional).
+func (rh *RsaHandler) Certificates() [][]byte {
+	return nil
+}
+
+// Internal data handlers
+
+func parseRsaPublicKey() (*rsa.PublicKey, error) {
+	if len(rsaPubKeys) < 1 {
+		return nil, fmt.Errorf("no public keys")
+	}
+	// Using the first one available so far
+	block, rest := pem.Decode([]byte(rsaPubKeys[0]))
+	if len(rest) > 0 {
+		log.Warnf("failed to parse the public key")
+		return nil, fmt.Errorf("failed to parse the public key")
+	}
+
+	parsedKey, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	switch parsedKey.(type) {
+	case *rsa.PublicKey:
+		break
+	default:
+		log.Warnf("cannot parse the public key")
+		return nil, fmt.Errorf("cannot parse the public key")
+	}
+	return parsedKey.(*rsa.PublicKey), nil
+}
+
+type RsaAuthData struct {
+	VoterId   []byte
+	Message   []byte
+	Signature []byte
+}
+
+// parseRsaAuthData transforms the incoming authData string array and returns a digested output
+// of the relevant parameters for the handler
+func parseRsaAuthData(authData []string) (*RsaAuthData, error) {
+	result := new(RsaAuthData)
+
+	if len(authData) != 3 {
+		log.Warnf("invalid params (3 items expected)")
+		return nil, fmt.Errorf("invalid params (3 items expected)")
+	}
+
+	// Catenate hex
+	processId := authData[0]
+	if len(processId) != ELECTION_ID_STR_LENGTH {
+		log.Warnf("invalid electionId")
+		return nil, fmt.Errorf("invalid electionId")
+	}
+	voterId := authData[1]
+	if len(voterId) != VOTER_ID_STR_LENGTH {
+		log.Warnf("invalid voterId")
+		return nil, fmt.Errorf("invalid voterId")
+	}
+
+	voterIdBytes, err := hex.DecodeString(voterId)
+	if err != nil || len(voterIdBytes) != VOTER_ID_STR_LENGTH/2 {
+		log.Warnf("invalid voterId: %s", voterId)
```

Comment:
I'd remove all these log.Warnf() messages when the error is returned.
The HTTP router layer, if logLevel set to "Debug", will print the responses and we'll see the errors returned. No need to print them duplicated.